### PR TITLE
Relax top level virtual service to allow multiple model names

### LINKED ISSUE #664: transform model_name and PVC are not working
Problem 1) 
I try to set transform model name but look like is not implemented
Bellow is my yaml: 
`apiVersion: "serving.kubeflow.org/v1alpha2"
kind: "InferenceService"
metadata:
  name: "my-mnist-serving"
  namespace: "kfserving"
spec:
  default:
    predictor:
      minReplicas: 1
      tensorflow:
        storageUri: "pvc://kfserving-pvc/models/mnist"
    transformer:
      minReplicas: 1
      custom:
        container:
          image: my-image
          name: kfserving-container
          args:
          - --model_name
          - mnist
          env:
          - name: STORAGE_URI
            value: pvc://kfserving-pvc/models`

As you can se I specify model_name as mnist but when I look in logfile after container is created I can see this: "kfserver:83] Registering model: my-mnist-serving" 
Looking in your code here: https://github.com/kubeflow/kfserving/blob/66e03f70f8615185015d6380a2d71ba7aa5e5470/pkg/apis/serving/v1alpha2/transformer.go#L43 I can see that model name is inferenceservice name which is fine.
but since I pass  --model_name it must get that instead inferenceservice name or --model_name must override inferenceservice name

Problem 2)
I try to set PVC in my inferenceservice like for predict custom image but is not working
As you can see in up yaml I do it like this:
`env:
    - name: STORAGE_URI
       value: pvc://kfserving-pvc/models``
From some reasons volume is not created. I suspect that is not implemented also.

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:
We should not assume service name is same as the model name, this is also important when we want to support model sharing, so user can predict on a given model which is specified on the endpoint path.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #664 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/667)

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -26,3 +26,10 @@ Note: Only Tensorflow models support the fields "signature_name" and "inputs".
 
 ## Explain
 All InferenceServices that are deployed with an Explainer support a standardized explanation API. This interface is identical to the Tensorflow V1 HTTP API with the addition of an ":explain" verb.
+
+## Endpoints
```
Comment (ellis-bigelow) [inline docs/README.md]: Can we put this table directly in the Data Plane header?

**Comment by yuzisun** [comment]
> I think I disagree with the premise here. There's some really nice simplifications that come with treating modelname and service name equivalently. I don't see this as a sharing blocker either, as the definitions will still be separate

we kind of enforcing model name and service name to be the same for OOTB predictors, however this breaks when user goes to custom path with custom transformer or predictor, user can specify `--model-name` and service name differently and this breaks the data plane.

**Comment by ellis-bigelow** [comment]
What if we, by default, injected a set of environment variables (like ModelName). We could build our images to bind those environment variables to arguments. That would allow users to deterministicly rely on those environment variables and not have to specify it themselves.

KF_MODEL_NAME="abc"

args: {
 --model_name=$KF_MODEL_NAME
}

**Comment by yuzisun** [comment]
@ellis-bigelow Can you take a look at this? This is to relax the top level virtual service to allow multiple models, it was restricted to match with the service name and now I changed to be a regex.

**Comment by ellis-bigelow** [comment]
Cool. Thanks for the fix!

**Pull request merged**